sanic/request.py

```diff
@@ -519,8 +519,11 @@ def url_for(self, view_name, **kwargs):
         :rtype: str
         """
         # Full URL SERVER_NAME can only be handled in app.url_for
-        if "//" in self.app.config.SERVER_NAME:
-            return self.app.url_for(view_name, _external=True, **kwargs)
+        try:
+            if "//" in self.app.config.SERVER_NAME:
+                return self.app.url_for(view_name, _external=True, **kwargs)
+        except AttributeError:
+            pass
```

Comment:
What is this expected to do? I gather that full URL will still be mishandled by the logic that follows.